pkg/controller/machinepool/awsactuator_test.go

```diff
@@ -453,6 +454,39 @@ func TestAWSActuator(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
I'm going to need to stare at this test case for a minute.

Meanwhile: Do we need to add test cases for the error paths you've introduced?

Reply:
I could do error path tests for no subnets configured (invalid) or no subnets returned. This test case also needs to validate the filter, adding that.

Reply:
Added the validation piece to this test which is checking the filters against expected.

Reply:
Added a few error path tests and made the other pieces line up.